test/integration/coscheduling_test.go

```diff
@@ -34,16 +45,69 @@ import (
 	testutils "k8s.io/kubernetes/test/integration/util"
 	imageutils "k8s.io/kubernetes/test/utils/image"
 
+	scheconfig "sigs.k8s.io/scheduler-plugins/pkg/apis/config"
+	"sigs.k8s.io/scheduler-plugins/pkg/apis/scheduling/v1alpha1"
 	"sigs.k8s.io/scheduler-plugins/pkg/coscheduling"
+	pgclientset "sigs.k8s.io/scheduler-plugins/pkg/generated/clientset/versioned"
+	coschedulingutil "sigs.k8s.io/scheduler-plugins/pkg/util"
 	"sigs.k8s.io/scheduler-plugins/test/util"
-	// Ensure scheme package is initialized.
-	_ "sigs.k8s.io/scheduler-plugins/pkg/apis/config/scheme"
 )
 
-var lowPriority, midPriority, highPriority = int32(0), int32(100), int32(1000)
-
 func TestCoschedulingPlugin(t *testing.T) {
+	todo := context.TODO()
+	ctx, cancelFunc := context.WithCancel(todo)
+	testCtx := &testutils.TestContext{
+		Ctx:      ctx,
+		CancelFn: cancelFunc,
+		CloseFn:  func() {},
+	}
 	registry := fwkruntime.Registry{coscheduling.Name: coscheduling.New}
+	t.Log("create apiserver")
+	_, config := util.StartApi(t, todo.Done())
+
+	config.ContentType = "application/json"
+
+	apiExtensionClient, err := apiextensionsclient.NewForConfig(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	kubeConfigPath := ""
+	if tempFile, err := ioutil.TempFile(os.TempDir(), "kubeconfig-"); err == nil {
+		kubeConfig := createKubeConfig(config)
+		clientcmd.WriteToFile(*kubeConfig, tempFile.Name())
+		kubeConfigPath = tempFile.Name()
+	}
+	defer os.RemoveAll(kubeConfigPath)
+
+	t.Log("create crd")
+	crd := &apiextensionsv1beta1.CustomResourceDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "podgroups.scheduling.sigs.k8s.io",
+		},
+		Spec: apiextensionsv1beta1.CustomResourceDefinitionSpec{
+			Group:   "scheduling.sigs.k8s.io",
+			Version: "v1alpha1",
+			Scope:   apiextensionsv1beta1.NamespaceScoped,
+			Names: apiextensionsv1beta1.CustomResourceDefinitionNames{
+				Kind:       "PodGroup",
+				Plural:     "podgroups",
+				ShortNames: []string{"pg", "pgs"},
+			},
+		},
+	}
+
+	_, err = apiExtensionClient.ApiextensionsV1beta1().CustomResourceDefinitions().Create(ctx, crd, metav1.CreateOptions{})
+
+	if err != nil {
+		t.Fatal(err)
+	}
+	time.Sleep(1 * time.Second)
```

Comment:
We should avoid time.Sleep() if we have a way to ensure deterministic behavior.

Here I guess you're trying to ensure the CRD is created and ready to serve CR requests?

Reply:
Yes